spec/std/string_spec.cr

```diff
@@ -773,6 +773,24 @@ describe "String" do
     it { "hello\n\n\n\n".chomp("").should eq("hello\n\n\n\n") }
 
     it { "hello\r\n".chomp("\n").should eq("hello") }
+
+    it "pre-computes string size if possible" do
+      {"!hello!", "\u{1f602}hello\u{1f602}", "\xFEhello\xFF"}.each do |str|
+        {"", "\n", "\r", "\r\n"}.each do |newline|
+          x = str + newline
+          x.size # side-effect
+          y = x.chomp
+          y.@length.should eq(7)
+        end
+      end
+    end
+
+    it "does not pre-compute string size if not possible" do
+      x = String.build &.<< "abc\n"
+      x.@length.should eq(0)
```

Comment:
**suggestion:** Use `String#size_known?` instead. It puts a clear label. It's not a public API method, but that should be fine for a unit test.
```suggestion
      x.size_known?.should be_true
```
Ditto below.